config-model/src/test/java/com/yahoo/config/model/provision/ModelProvisioningTest.java

```diff
@@ -1356,48 +1356,109 @@ public void testMultitenantButNotHosted() {
         assertEquals(1, controller.getContainers().size());
     }
 
+    @Test
+    public void testSharedNodesNotHosted() {
+        String hosts =
+                "<?xml version=\"1.0\" encoding=\"utf-8\" ?>\n" +
```

Comment:
Using the joinLines() helper, you wouldn't have to type all those newlines.

Reply:
The newlines are actually unnecessary ... I didn't type them though: IntelliJ automatically makes this code when you paste a multilevel string, which is what I did here (source was a gh issue).